src/MockGenerator/Program.cs

```diff
@@ -0,0 +1,372 @@
+        {
+            var baseFileName = Path.GetFileNameWithoutExtension(file);
+
+            fileDirectory = Path.GetDirectoryName(file);
+            Debug.Assert(fileDirectory != null, "fileDirectory != null");
+            return baseFileName;
+        }
+
+        private static InterfaceDeclarationSyntax DecorateInterfaceWithWrapperFunction(
+            MethodDeclarationSyntax methodDeclaration,
+            IdentifierNameSyntax invokeMethodIdentifier,
+            InterfaceDeclarationSyntax interfaceDeclaration)
+        {
+            var dllImport = methodDeclaration.AttributeLists
+                .First(x => x.OpenBracketToken.HasLeadingTrivia);
+            var interfaceMethodDeclaration = SyntaxFactory.MethodDeclaration(
+                SyntaxFactory.List<AttributeListSyntax>(),
+                SyntaxFactory.TokenList(),
+                methodDeclaration.ReturnType,
+                default(ExplicitInterfaceSpecifierSyntax),
+                invokeMethodIdentifier.Identifier,
+                methodDeclaration.TypeParameterList,
+                methodDeclaration.ParameterList,
+                methodDeclaration.ConstraintClauses,
+                default(BlockSyntax),
+                SyntaxFactory.Token(SyntaxKind.SemicolonToken));
+
+            interfaceDeclaration = interfaceDeclaration
+                .AddMembers(
+                    interfaceMethodDeclaration
+                        .WithTrailingTrivia(
+                            NewLineCharacter,
+                            TabCharacter)
+                        .WithLeadingTrivia(dllImport.OpenBracketToken.LeadingTrivia));
+            return interfaceDeclaration;
+        }
+
+        private static ClassDeclarationSyntax DecorateClassWithWrapperFunction(MethodDeclarationSyntax methodDeclaration,
+            IdentifierNameSyntax invokeMethodIdentifier,
+            ClassDeclarationSyntax classDeclaration)
+        {
+            var dllImport = methodDeclaration.AttributeLists
+                .First(x => x.OpenBracketToken.HasLeadingTrivia);
+            var arguments = methodDeclaration.ParameterList
+                .Parameters
+                .Select((x, i) =>
+                {
+                    var identifierName = SyntaxFactory.Argument(
+                        null,
+                        x.Modifiers.FirstOrDefault(z => z.IsKind(SyntaxKind.RefKeyword) || z.IsKind(SyntaxKind.OutKeyword)),
+                        SyntaxFactory.IdentifierName(x.Identifier));
+                    if (i > 0)
+                    {
+                        identifierName = identifierName.WithLeadingTrivia(WhitespaceCharacter);
+                    }
+
+                    return identifierName;
+                });
+
+            var arrowBody = SyntaxFactory.ArrowExpressionClause(
+                SyntaxFactory.Token(SyntaxKind.EqualsGreaterThanToken)
+                    .WithLeadingTrivia(
```

Comment:
Ah nice! ReSharper also offers the free `InspectCode.exe` now. Not sure if it can actually bulk fix files, but that could be relevant as a build step.